web/src/main/java/com/epam/esm/web/controller/UserController.java

```diff
@@ -0,0 +1,104 @@
+package com.epam.esm.web.controller;
+
+import com.epam.esm.model.dto.*;
+import com.epam.esm.service.PurchaseService;
+import com.epam.esm.service.UserService;
+import com.epam.esm.service.exeption.RecourseNotExistException;
+import com.epam.esm.web.utils.HateoasWrapper;
+import lombok.AllArgsConstructor;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.validation.annotation.Validated;
+import org.springframework.web.bind.annotation.*;
+
+import javax.validation.Valid;
+import javax.validation.constraints.Min;
+import javax.validation.constraints.Positive;
+import java.util.List;
+
+import static org.springframework.util.MimeTypeUtils.APPLICATION_JSON_VALUE;
+
+/**
+ * The  User controller with mapping "/users"
+ * It used for Rest Api
+ *
+ * @author Katerina Charakhovich
+ * @version 1.0.0
+ */
+@RestController
+@AllArgsConstructor
+@RequestMapping(value = "/users", produces = APPLICATION_JSON_VALUE)
+@Validated
+public class UserController {
+    private static final String DEFAULT_OFFSET = "0";
+    private static final String DEFAULT_LIMIT = "10";
+    private static final String INVALID_OFFSET_MESSAGE = "invalid value parameter offset";
+    private static final String INVALID_LIMIT_MESSAGE = "invalid value parameter limit";
+    private static final String OFFSET = "offset";
+    private static final String LIMIT = "limit";
+    @Autowired
+    UserService userService;
```

Comment:
please use `private` access level everywhere
use another access level only if it really needed

Reply:
Done